### PR TITLE
CUDA setup.hpp fixes

### PR DESCRIPTION
**Pull request by hrehfeld**
I tried to fix setup.hpp so cuda detects some features, but I'm not sure it's correct. 

CUDA uses host compilers, so in theory you would need to detect host compiler features, but the CUDA compiler also has some extra quirks...

### CODE REVIEW

File: glm/detail/setup.hpp [hunk 1]
```diff
@@ -709,7 +711,8 @@
 #else
 #	define GLM_HAS_INITIALIZER_LISTS ((GLM_LANG & GLM_LANG_CXX0X_FLAG) && (\
 		((GLM_COMPILER & GLM_COMPILER_GCC) && (GLM_COMPILER >= GLM_COMPILER_GCC44)) || \
-		((GLM_COMPILER & GLM_COMPILER_VC) && (GLM_COMPILER >= GLM_COMPILER_VC2013))))
+		((GLM_COMPILER & GLM_COMPILER_VC) && (GLM_COMPILER >= GLM_COMPILER_VC2013)) || \
+		((GLM_COMPILER & GLM_COMPILER_CUDA))))
```
Comment (Groovounet) [inline glm/detail/setup.hpp]: From which version is this supported?

File: glm/detail/setup.hpp [hunk 2]
```diff
@@ -731,7 +734,8 @@
 #	define GLM_HAS_DEFAULTED_FUNCTIONS ((GLM_LANG & GLM_LANG_CXX0X_FLAG) && (\
 		((GLM_COMPILER & GLM_COMPILER_GCC) && (GLM_COMPILER >= GLM_COMPILER_GCC44)) || \
 		((GLM_COMPILER & GLM_COMPILER_VC) && (GLM_COMPILER >= GLM_COMPILER_VC2013)) || \
-		((GLM_COMPILER & GLM_COMPILER_INTEL) && (GLM_COMPILER >= GLM_COMPILER_INTEL12))))
+		((GLM_COMPILER & GLM_COMPILER_INTEL) && (GLM_COMPILER >= GLM_COMPILER_INTEL12)) || \
+		(GLM_COMPILER & GLM_COMPILER_CUDA)))
```
Comment (Groovounet) [inline glm/detail/setup.hpp]: From which version is this supported?

File: glm/detail/setup.hpp [hunk 3]
```diff
@@ -674,6 +674,7 @@
 #else
 #	define GLM_HAS_STATIC_ASSERT ((GLM_LANG & GLM_LANG_CXX0X_FLAG) && (\
 		((GLM_COMPILER & GLM_COMPILER_GCC) && (GLM_COMPILER >= GLM_COMPILER_GCC43)) || \
+		((GLM_COMPILER & GLM_COMPILER_CUDA)) || \
```
Comment (Groovounet) [inline glm/detail/setup.hpp]: From which version is this supported?

**Comment by hrehfeld** [comment]
I don't know, and I can't invest more time digging through nvcc docs. 

Frankly, CUDA support atm is just broken without these fixes, so I'd argue for either
a) just accept it as its better than nothing (maybe slap CUDA 7.5 detection on it)
b) if you want to support old CUDA versions, someone needs to dig through the docs. But as I said, CUDA is complicated, as it uses the host compiler, but still has its own compilation quirks.

Maybe someone from NVIDIA/nvcc can help out?

**Comment by Groovounet** [comment]
This issue should be fixed in master branch.

Thanks for reporting!
Christophe

**Pull request merged**